### PR TITLE
Refactor read only middleware

### PR DESCRIPTION
**Pull request by maarcingebala**
This PR refactors `ReadOnlyMiddleware`:
- reimplemented as Graphene middleware instead of Django's which makes it a bit simpler
- removed outdated HTML template and checking Django paths (no longer needed since we've dropped static views)
- added `ROOT_EMAIL` feature (which was used on `demo` branch but not moved to `master`)
- removed deprecated meta mutations from `ALLOWED_MUTATIONS` (I think it's not necessary to allow users to test them in demo mode).

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/middleware.py [hunk 1]
```diff
@@ -58,6 +59,51 @@ def app_middleware(next, root, info, **kwargs):
     return next(root, info, **kwargs)
 
 
+class ReadOnlyMiddleware:
+    ALLOWED_MUTATIONS = [
```
Comment (NyanKiyoshi) [inline saleor/graphql/middleware.py:63]: I noticed we don't allow users to register new accounts. Is it meant?

File: saleor/demo/settings.py [hunk 2]
```diff
@@ -81,3 +81,6 @@ def before_send(event: dict, _hint: dict):
     sentry_sdk.init(
         DEMO_SENTRY_DSN, integrations=[DjangoIntegration()], before_send=before_send,
     )
+
+
+ROOT_EMAIL = os.environ.get("ROOT_EMAIL")
```
Comment (NyanKiyoshi) [inline saleor/demo/settings.py:86]: We already have it on line 43

File: saleor/demo/settings.py [hunk 3]
```diff
@@ -81,3 +81,6 @@ def before_send(event: dict, _hint: dict):
     sentry_sdk.init(
         DEMO_SENTRY_DSN, integrations=[DjangoIntegration()], before_send=before_send,
```
Comment (NyanKiyoshi) [inline saleor/demo/settings.py:82]: We need to enable Celery too

File: saleor/graphql/middleware.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/middleware.py:63]: Yes it is, we don't want them to create accounts on demo.

File: saleor/demo/settings.py (hunk 3 above)
Comment (maarcingebala) [reply saleor/demo/settings.py]: Fixed

File: saleor/demo/settings.py (hunk 2 above)
Comment (maarcingebala) [reply saleor/demo/settings.py]: Good catch, fixed.

**Pull request merged**